### PR TITLE
add dropdown component

### PR DESCRIPTION
**Pull request by olegshilov**

### CODE REVIEW

File: src/components/Dropdown/DropdownMenuItem.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  shouldComponentUpdate(nextProps) {
+    return nextProps.children !== this.props.children ||
+           nextProps.className !== this.props.className;
+  }
+
+  handleMenuItemClick(event) {
+    const { onClick } = this.props;
+
+    if (onClick) {
+      onClick(event);
```
Comment (nkt) [inline src/components/Dropdown/DropdownMenuItem.js]: pass `this.props.onClick` directly to `div`

File: src/components/Dropdown/Dropdown.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+}
+
+.divider {
+  display: inline-block;
+  width: 100%;
+  box-sizing: border-box;
+  position: relative;
+  line-height: 0;
+  padding: var(--dropdown-padding-top) var(--dropdown-padding-right) var(--dropdown-padding-bottom) var(--dropdown-padding-left);
```
Comment (nkt) [inline src/components/Dropdown/Dropdown.css]: Split it to `padding-<side>` syntax

File: src/components/Dropdown/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+class Dropdown extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.node.isRequired,
+    isOpen: PropTypes.bool.isRequired
+  };
+
+  shouldComponentUpdate(nextProps) {
+    return nextProps.children !== this.props.children ||
+           nextProps.className !== this.props.className;
```
Comment (nkt) [inline src/components/Dropdown/Dropdown.js]: Add missing props

File: src/components/Dropdown/DropdownMenuItem.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import styles from './Dropdown.css';
+
+class DropdownMenuItem extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.node.isRequired,
+    divider: PropTypes.bool,
```
Comment (nkt) [inline src/components/Dropdown/DropdownMenuItem.js]: Remove this and create `DropdownMenuDivider` component.

File: src/components/Dropdown/DropdownMenuItem.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  constructor(props) {
+    super(props);
+
+    this.handleMenuItemClick = this.handleMenuItemClick.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return nextProps.children !== this.props.children ||
+           nextProps.className !== this.props.className;
```
Comment (nkt) [inline src/components/Dropdown/DropdownMenuItem.js]: Add missing props

File: src/styles/variables.css [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
   --default-font-size: 16px;
   --large-font-size: 20px;
   --default-line-height: 24px;
   --default-padding: 1rem;
-  --default-border-radius: 2px;
+  --default-border-radius: 5px;
   --spinner-max-height: 20px;
+
+  /* Layers */
+  --z-index-dropdown: 10;
```
Comment (nkt) [inline src/styles/variables.css]: :+1:

File: src/components/Dropdown/Dropdown.css [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  --dropdown-border-radius: var(--default-border-radius);
+  --dropdown-z-index: var(--z-index-dropdown);
+  --dropdown-item-height: 37px;
+  --dropdown-item-hover-background: color(#000 alpha(10%));
+  --dropdown-divider-color: color(#000 alpha(15%));
+  --dropdown-min-width: 200px;
+}
+
+.root {
+  all: initial;
```
Comment (nkt) [inline src/components/Dropdown/Dropdown.css]: Move all `all` props on top of file. I mean:
```css
. root, .bar, .baz {
  all: initial;
}
```

File: src/components/Dropdown/Dropdown.css [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+.item {
+  all: initial;
+  font-family: var(--dropdown-font-family);
+  font-size: var(--dropdown-font-size);
+  line-height: var(--dropdown-item-height);
+  display: table-row;
+  width: 100%;
+  box-sizing: border-box;
+
+  & span {
```
Comment (nkt) [inline src/components/Dropdown/Dropdown.css]: You should use classes instead of tags, when possible.

File: src/components/Dropdown/Dropdown.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.node.isRequired,
+    isOpen: PropTypes.bool.isRequired
+  };
+
+  shouldComponentUpdate(nextProps) {
+    return nextProps.children !== this.props.children ||
+           nextProps.className !== this.props.className ||
+           nextProps.isOpen !== this.props.isOpen;
```
Comment (nkt) [inline src/components/Dropdown/Dropdown.js]: `isOpen` should be first, then `children`, then `className`.

File: src/components/Dropdown/DropdownMenuDivider.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import styles from './Dropdown.css';
+
+class DropdownMenuDivider extends Component {
```
Comment (nkt) [inline src/components/Dropdown/DropdownMenuDivider.js]: Maybe just `DropdownDivider`?

File: src/components/Dropdown/DropdownMenuItem.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import styles from './Dropdown.css';
+
+class DropdownMenuItem extends Component {
```
Comment (nkt) [inline src/components/Dropdown/DropdownMenuItem.js]: Maybe just `DropdownItem`?

File: src/components/Dropdown/Dropdown.js (hunk 9 above)
Comment (olegshilov) [inline src/components/Dropdown/Dropdown.js]: maybe we should update our component template?

**Pull request merged**